### PR TITLE
Cloud access role fix

### PR DESCRIPTION
**Pull request by Sugar-pack**
Fix a bug when there are multiple roles with the same ARN.
Fix a bug when a role with an empty ARN changes its atlasAssumedRoleExternalId after editing.
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

**Review by fabritsius — APPROVED** [review]
Looks Good 👍

File: pkg/controller/atlasproject/cloud_provider_access.go [hunk 1]
```diff
@@ -33,7 +33,16 @@ func ensureProviderAccessStatus(ctx context.Context, customContext *workflow.Con
...
 	client := customContext.Client
 	logger := customContext.Log
-	specToStatusMap := checkStatuses(specs, statuses)
+	specToStatusMap, haveDuplicate, cantMatch := checkStatuses(specs, statuses)
+	if haveDuplicate {
+		return workflow.Terminate(workflow.ProjectCloudAccessRolesIsNotReadyInAtlas, "some roles contains same ARN value"), status.CloudProviderAccessReadyType
+	}
+	if cantMatch {
+		return workflow.Terminate(workflow.ProjectCloudAccessRolesIsNotReadyInAtlas, "More than one new role "+
+			"with ARN may correspond to an existing empty role. Keep only one new role. If one of the new roles has been "+
```
Comment (igor-karpukhin) [inline pkg/controller/atlasproject/cloud_provider_access.go:42]: @Sugar-pack consider to change the message: "More than one new role with ARN may correspond to an existing empty role. Keep only one new role containing data from the status field and delete all other roles. You can add them again after authorization is complete."

Comment (Sugar-pack) [reply pkg/controller/atlasproject/cloud_provider_access.go]: Done

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**